infrastructure/expected-fail/assert_implements_optional.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```

Comment:
I think the logical place to put this kind of test is under `resources/tests` (although those are sadly more work to write an run, but they are also enabled in CI so just as effective for preventing regressions)

Reply:
> I think the logical place to put this kind of test is under resources/tests (although those are sadly more work to write an run, but they are also enabled in CI so just as effective for preventing regressions)

That looks like exactly what I want, awesome!

qq; it looks like `functional/` exists to test harness failure cases, but there are also `unit/` tests like `assert_object_equals.html` which use the fact that `assert()` throws an `AssertionError` to create failure tests without needing a comparison output:

```
function test_failure(fn, name) {
  test(function() {
    try {
      fn();
    } catch (err) {
      if (err instanceof AssertionError) {
        return;
      }
      throw new AssertionError('Expected an AssertionError, but');
    }
    throw new AssertionError(
      'Expected an AssertionError, but no error was thrown'
    );
  }, name);
}
```

What do you think of this pattern? I quite like it, but don't want to put the effort into using it if you're then going to ask me to switch to `functional/` for the failing tests :). (And I'm happy to use `functional/` if that's what we prefer).

Reply:
I don't have a strong preference. I think the format in `functional` turned out to be over-verbose, so I can understand not wanting to use that, but it is arguably more consistent. Up to you.

Reply:
Done. PTAL :)